lib/HLSL/HLSignatureLower.cpp

```diff
@@ -1145,10 +1145,12 @@ void HLSignatureLower::GenerateDxilInputsOutputs(DXIL::SignatureKind SK) {
       bI1Cast = true;
       Ty = i32Ty;
     }
-    if (!hlslOP->IsOverloadLegal(opcode, Ty)) {
+  
+    if (!Ty || !hlslOP->IsOverloadLegal(opcode, Ty)) {
       std::string O;
       raw_string_ostream OSS(O);
-      Ty->print(OSS);
+      if (Ty)
+        Ty->print(OSS);
```

Comment:
This is definitely an improvement to crashing. It would be better if we could print out the type even though it's not one of the basic types. I realize that information may not be readily available at this location. Perhaps it can be caught when the signature element for the texture type is created?

We definitely need at least one, probably a few tests for this. I suggest trying Textures, buffers, samplers, any HLSL types. Those tests will remain useful indefinitely.